### PR TITLE
kata-ctl: check: only-list-releases and include-all-releases options

### PR DESCRIPTION
**Pull request by alexlee-23**
This pull request adds kata-ctl check **only-list-releases** and **include-all-releases**

Fixes: [#5345](https://github.com/kata-containers/kata-containers/issues/5345)

Signed-off-by: Alex Lee [alee23@bu.edu](mailto:alee23@bu.edu)

### CODE REVIEW

**Comment by liubin** [comment]
Some merge commit in this PR, maybe a rebase is needed.

File: src/tools/kata-ctl/src/check.rs [hunk 1]
```diff
@@ -7,12 +7,41 @@
...
+struct Release {
+    #[serde(rename = "tag_name")]
+    tag_name: String,
+    #[serde(rename = "prerelease")]
+    prerelease: bool,
+    #[serde(rename = "created_at")]
+    created_at: String,
+    #[serde(rename = "tarball_url")]
+    tarball: String,
+    // #[serde(rename = "assets")]
```
Comment (Ji-Xinyou) [inline src/tools/kata-ctl/src/check.rs:22]: remove the comments please

File: src/tools/kata-ctl/src/ops/version.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 Intel Corporation
```
Comment (Ji-Xinyou) [inline src/tools/kata-ctl/src/ops/version.rs:1]: This entire file should be auto generated, do not edit on this.

**Review by jodh-intel — COMMENTED** [review]
Thanks @alexlee-23 ! A few comments...

/cc @dborquez, @amshinde, @cmaf for input.

**Review by jodh-intel — COMMENTED** [review]
Thanks @alexlee-23  - a few more comments.

btw, I mentioned that the golang `kata-runtime check` binary should not be listing the vendor files, so neither should `kata-ctl`.

For reference, the issue is: https://github.com/kata-containers/kata-containers/issues/3272

**Comment by alexlee-23** [comment]
Hello all, I tried to fix issues raised in the comments, not sure if there are any other issues. Please review my updated PR. Thank you!

**Review by jodh-intel — APPROVED** [review]
Thanks @alexlee-23 !

lgtm

File: src/tools/kata-ctl/src/check.rs [hunk 3]
```diff
@@ -144,6 +180,41 @@ pub fn check_version() -> Result<()> {
     Ok(())
 }
 
+pub fn check_all_releases() -> Result<()> {
+    let releases: Vec<Release> = get_kata_all_releases_by_url().map_err(handle_reqwest_error)?;
+
+    for release in releases {
+        if !release.prerelease {
+            println!(
+                "Official:   Release {:15}; created {} ; {}",
```
Comment (studychao) [inline src/tools/kata-ctl/src/check.rs:189]: nit: The white space number is different from `PreRelease: Release {:15}; created {} ; {}`, could you make it consistent?

File: src/tools/kata-ctl/src/check.rs [hunk 4]
```diff
@@ -107,15 +128,30 @@ pub fn run_network_checks() -> Result<()> {
...
+    let releases: Vec<Release> = reqwest::blocking::Client::new()
+        .get(KATA_GITHUB_RELEASE_URL)
+        .header(CONTENT_TYPE, JSON_TYPE)
+        .header(USER_AGENT, USER_AGT)
+        .send()?
+        .json()?;
+    Ok(releases)
+}
+
+#[cfg(any(
```
Comment (studychao) [inline src/tools/kata-ctl/src/check.rs:150]: why adding a `target_arch` judegemt here?

**Review by studychao — COMMENTED** [review]
Thanks @alexlee-23 , nice work ! Just 2 little comments, please take a look

**Comment by alexlee-23** [comment]
> Thanks @alexlee-23 , nice work ! Just 2 little comments, please take a look

Thank you @studychao, I hope these changes are ok!

**Review by studychao — APPROVED** [review]
LGTM, good work ! @alexlee-23

**Comment by studychao** [comment]
@alexlee-23 
For you new commit ([Fixed format](https://github.com/kata-containers/kata-containers/pull/5786/commits/d281abf3b8a82723f2cd76e10eb2293cc245ad32)) please also follow the commit format.

**Comment by alexlee-23** [comment]
Great! Thank you @studychao I will get that fixed asap.

**Pull request merged**